pyanaconda/nm.py

```diff
@@ -961,6 +969,38 @@ def nm_delete_connection(uuid):
     proxy = _get_proxy(object_path=settings_paths[0], interface_name="org.freedesktop.NetworkManager.Settings.Connection")
     proxy.Delete()
 
+
+def nm_update_settings(uuid, new_values):
+    """Update settings of connection given by uuid.
+
+       The type of value is determined from existing settings of device.
+       If setting for key1, key2 does not exist, default_type_str is used or
+       if None, the type is inferred from the value supplied (string and bool only).
+
+       :param uuid: uuid of the connection
+       :type name: str
+       :param new_values: list of settings with new values and its types
+                          [[key1, key2, value, default_type_str]]
+                          key1: first-level key of setting (eg "connection")
+                          key2: second-level key of setting (eg "uuid")
+                          value: new value
+                          default_type_str: dbus type of new value to be used
+                                            if the setting does not already exist;
+                                            if None, the type is inferred from
+                                            value (string and bool only)
+       :type new_values: [[key1, key2, value, default_type_str], ...]
+                         key1: str
+                         key2: str
+                         value:
+                         default_type_str: str
```

Comment:
I'm not sure if this will be build correctly as the documentation. It would be nice to look on it before pushing the changes to avoid non-readable documentation.

Another thing is, wouldn't it be better to create object which will contain `key1`, `key2`, `value` and `default_type_str`. It could also handle validation and it will be easy to understand what to pass where.
I'm not sure if this is a right step now, because I think this is used somewhere else already. However it could be worth the effort even for future changes.